addons/base_automation/models/base_automation.py

```diff
@@ -876,7 +886,8 @@ def write(self, vals, **kw):
                     return write.origin(self, vals, **kw)
                 records = self.with_env(automations.env).filtered('id')
                 # check preconditions on records
-                pre = {a: a._filter_pre(records) for a in automations}
+                changed_fields = [self._fields[f] for f in vals]
+                pre = {a: a._filter_pre(records, changed_fields=changed_fields) for a in automations}
```

Comment:
AFAIU, that's useless to do this in the write since the `write.origin` will be add_to_recompute the right records.

Reply:
Indeed, this is not needed since `modified` will mark them for recomputation.